deltachat-rpc-client/tests/test_something.py

```diff
@@ -846,3 +846,35 @@ def test_delete_deltachat_folder(acfactory, direct_imap):
     assert msg.text == "hello"
 
     assert "DeltaChat" in ac1_direct_imap.list_folders()
+
+
+def test_broadcast(acfactory):
+    alice, bob = acfactory.get_online_accounts(2)
+
+    alice_chat = alice.create_broadcast("My great channel")
+    snapshot = alice_chat.get_basic_snapshot()
+    assert snapshot.name == "My great channel"
+    assert snapshot.is_unpromoted
+    assert snapshot.is_encrypted
+    assert snapshot.chat_type == ChatType.OUT_BROADCAST
+
+    alice_contact_bob = alice.create_contact(bob, "Bob")
+    alice_chat.add_contact(alice_contact_bob)
+
+    alice_msg = alice_chat.send_message(text="hello").get_snapshot()
+    assert alice_msg.text == "hello"
+    assert alice_msg.show_padlock
+
+    bob_msg = bob.wait_for_incoming_msg().get_snapshot()
+    assert bob_msg.text == "hello"
+    assert bob_msg.show_padlock
+    assert bob_msg.error is None
+
+    bob_chat = bob.get_chat_by_id(bob_msg.chat_id)
+    bob_chat_snapshot = bob_chat.get_basic_snapshot()
+    assert bob_chat_snapshot.name == "My great channel"
+    assert not bob_chat_snapshot.is_unpromoted
+    assert bob_chat_snapshot.is_encrypted
+    assert bob_chat_snapshot.chat_type == ChatType.IN_BROADCAST
+
```

Comment:
Shouldn't it be checked that the broadcast chat is in a "Request" state and can be unblocked by Bob?